R/utilities-data-set.R

```diff
@@ -0,0 +1,46 @@
+
+makeColumn <- function(dataSet, property) {
+  if (length(dataSet[[property]]) == 0) {
+    rep(NA, length(dataSet$xValues))
+  } else if (length(dataSet[[property]]) == 1) {
+    rep(dataSet[[property]], length(dataSet$xValues))
+  } else {
+    dataSet[[property]]
+  }
+}
+
+# # retrieve entry 'property' from a DataSet object
+# makeXYColumn <- function(dataSet, property) {
+#   if (length(dataSet[[property]]) == 0) {
+#     rep(NA, length(dataSet$xValues))
+#   } else {
+#     dataSet[[property]]
+#   }
+# }
+
+#' Converts a list of DataSet objects to a data.frame
+#'
+#' @param dataSets A list of DataSet objects or a single DataSet
+#'
+#' @return DataSet objects as dataframe with columns name, xValue, yValue, yErrorValues,
+#' xDimension, xUnit, yDimension, yUnit, yErrorType, yErrorUnit, yMolWeight
+#' @export
+dataSetToDataFrame <- function(dataSets) {
```

Comment:
we need tests for this method, at least just to make sure that this does not crash with updated DataSet class